aml-development/ozp-backend
# feat(profile): add ability to update storefront customizations

AMLOS-743

**Description**     
As a user, I would like to customize the view of my storefront in my profile.  (Reorder/hide/display right side components - Frequently Visited, Updates, My Listings)

Possible starting point: https://github.com/aml-development/frontend/compare/customize_storefront

**Acceptance Criteria**   
The storefront view is customizable through the user profile.

This PR is for the backend functionality necessary for this feature
 
**How to test code**    
Pull `customize_storefront`

Get the user profile, ie:
```
GET http://localhost:8001/api/profile/{id}/
GET http://localhost:8001/api/self/profile/
```
In the payload, there should be a field called `storefront_customizations`, which should be an empty array.

Update the storefront customizations, using the following API call:
```
PUT http://localhost:8001/api/profile/{id}/
```
An example payload is as follows:
```
        {
            "storefront_customizations": [
                {"section": "FEATURED", "is_hidden": true},
                {"section": "RECOMMENDED", "position": 1},
                {"section": "RECENTLY_ADDED", "position": 2, "is_hidden": false},
                {"section": "MOST_POPULAR", "position": 3, "is_hidden": true},
                {"section": "FREQUENTLY_VISITED"},
                {"section": "UPDATES", "is_hidden": false},
                {"section": "MY_LISTINGS", "position": 4, "is_hidden": false}
            ]
        }
```
Each item has a mandatory field `section`, a field for `position` (default `0`) and a field for `is_hidden` (default `false`).  

**Please Review**     
@aml-development/ozp-backend-team    

**Checklist**
